app/decorators/order_meal_decorator.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class OrderMealDecorator < ApplicationDecorator
+  delegate :meal_type,        to: :food_item, allow_nil: true
+  delegate :food_item, :menu, to: :meal,      allow_nil: true
+
+  def related_food_items
+    menu.meals
+      .joins(:food_item)
+      .where(food_items: { meal_type: meal_type })
+      .pluck('food_items.name, meals.id')
+  end
+
+  def menu
+    Menu.last
```

Comment:
that's some pretty weird solution. Shouldn't you manipulate some particular menu?
Users can create A LOT of Menus at once, it's not always the last one you wanna get.